src/lib.rs

```diff
@@ -225,10 +225,25 @@ impl FromStr for WitnessProgram {
             bech32: s.to_string(),
         };
         wp.validate()?;
+        if let Some(version_variant) = program_version_to_variant(version) {
+            if version_variant != variant {
+                return Err(Error::InvalidEncoding);
+            }
+        }
```

Comment:
What happens if we parse an invalid version? As it is we don't check the variant at all. I think the else branch should return `Err(Error::InvalidScriptVersion)`.

Reply:
We call `wp.validate()?` before this final variant check, so we'll error out on script version and length prior to this point. I think we're following the [pseudo-code in BIP-0350](https://github.com/bitcoin/bips/blob/master/bip-0350.mediawiki#addresses-for-segregated-witness-outputs) pretty closely here, which performs the Bech32m check last.

Reply:
Ah, I see, totally missed that!